test/integration/users.test.js

```diff
@@ -694,6 +694,45 @@ describe("Users", function () {
     });
   });
 
+  describe("GET /users/username", function () {
+    const firstname = "shubham";
+    const lastname = "sigdar";
+
+    it("Should return unique username when passing firstname and lastname", function (done) {
+      chai
+        .request(app)
+        .get(`/users/username?firstname=${firstname}&lastname=${lastname}&dev=true`)
+        .set("cookie", `${cookieName}=${jwt}`)
+        .end((err, res) => {
+          if (err) {
+            return done();
+          }
+          expect(res).to.have.status(200);
+          expect(res.body).to.be.a("object");
+          expect(res.body.username).to.equal("shubham-sigdar");
+
+          return done();
+        });
+    });
+
```

Comment:
PLease write test for case when username is already present in DB

Reply:
Done with changes can you please check latest commit